### DISCUSSION TITLE
how to remove a field from query.

### DISCUSSION THREAD

**Discussion by amitnekar** [open-policy-agent/community#562] [OPA and Rego]
I am trying to set result["body"] := "sometext". But it is not reflecting in the output response. What could be the reson

**Comment by srenatus** [comment]
How so you query OPA? If you set a rule like that in package foo.bar, you should only get that when querying /v1/data/foo/bar/result.

**Reply by anderseknert** [reply]
Adding to Stephan's answer — if you're using the Rego Playground to evaluate the policy, it'll evaluate the whole package (including all rules) by default, and include the result of that in the output. You can also select the text of any individual rule which will have the "Evaluate" button changed to "Evaluate selection", which will output only the result of that single rule.

**Reply by amitnekar** [reply]
i would like to rephrase the question a bit more. now that i have tried a few more things.
in the opa rego file is there a way to change the request object. I am trying to change it thru code and its giving error cannot modify ref variables. I wanted to delete the query field from request object if the user is not authorized to view that field.
